File: tasks/vulcancyber-v1/oss-urllib3-host-injection-validation/tests/vb_regression.py
# Hidden pass_to_pass regression guard for oss-urllib3-host-injection-validation.
#
# Rejecting invalid hosts must not reject well-formed ones: an ordinary domain and
# an IPv4 literal still parse with the expected host. Both hold at the base commit
# and after the fix. Run with pytest.

from urllib3.util.url import parse_url


def test_ordinary_domain_parses():
    assert parse_url("http://example.com/path").host == "example.com"


def test_ipv4_host_parses():
    assert parse_url("http://127.0.0.1/path").host == "127.0.0.1"

File: tasks/vulcancyber-v1/oss-gix-validate-lone-at-refname/tests/tests/vb_regression.rs
// Hidden pass_to_pass regression guard for oss-gix-validate-lone-at-refname.
//
// Rejecting a lone "@" must not reject an "@" that appears inside a normal
// reference name, nor any ordinary name. Both hold at the base commit and after
// the fix. Run with `cargo test --offline`.

use bstr::ByteSlice;
use gix_validate::reference;

#[test]
fn at_inside_a_component_is_valid() {
    assert!(reference::name_partial(b"refs/heads/@".as_bstr()).is_ok());
}

#[test]
fn ordinary_name_is_valid() {
    assert!(reference::name_partial(b"refs/heads/main".as_bstr()).is_ok());
}
